### PR TITLE
Misc fixes.

### PR DESCRIPTION
**Pull request by fisx**

### CODE REVIEW

**Comment by fisx** [comment]
(There is a lot of stuff in here that i want to release now, but I'll leave the `please review` label on.)

File: src/Frontend/Page/Overview.hs [hunk 1]
```diff
@@ -174,26 +174,24 @@ instance ToHtml Tabs where
...
   deriving (Eq, Show, Read, Generic)
 
 data ListItemIdea = ListItemIdea
-      { _listItemIdeaContext    :: ListItemIdeaContext
-      , _listItemIdeaPhase      :: Maybe Phase
-      , _listItemIdeaNumVoters  :: Int
-      , _listItemIdea           :: Idea
-      , _listItemRenderContext  :: RenderContext
+      { _listItemRenderContext  :: RenderContext
+      , _listItemIdeaContext    :: ListItemIdeaContext
```
Comment (Mikolaj) [inline src/Frontend/Page/Overview.hs]: Perhaps Position? There is already ` data IdeaLocation`. Also, we don't want to mix it up with Space.

File: src/Persistent/Pure.hs [hunk 2]
```diff
@@ -245,6 +246,9 @@ runPersistExcept :: PersistExcept -> ServantErr
 runPersistExcept (PersistError500 msg)            = err500 { errBody = cs msg }
 runPersistExcept (PersistError404 msg)            = err404 { errBody = cs msg }
 runPersistExcept (PersistErrorNotImplemented msg) = err500 { errBody = cs msg }
+runPersistExcept (UserLoginInUse li) =
+    err500 { errBody = "user login in use: " <> cs (show li) }
+    -- FIXME: what's a good status code for 'login in use'?
```
Comment (Mikolaj) [inline src/Persistent/Pure.hs]: 200? USers are know to reload pages and go back, etc. Otherwise, perhaps 403:  https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.4

File: src/Action.hs [hunk 3]
```diff
@@ -353,18 +353,19 @@ voteIdeaCommentReply ideaId commentId replyId =
...
     idea  <- mquery $ findIdea iid
     topic <- mquery $ ideaTopic idea
-    -- FIXME: should this be one transaction?  or the two above as well?
     equery $ checkInPhase (PhaseJury ==) idea topic
     currentUserAddDb_ (AddIdeaJuryResult iid) rv
     checkCloseJuryPhase topic
 
 checkCloseJuryPhase :: ActionM m => Topic -> m ()
 checkCloseJuryPhase topic = do
+    -- FIXME: should this be one transaction?
```
Comment (Mikolaj) [inline src/Action.hs]: I think so, and the same above. I think an alternative is to check (in the operations above that modify the DB, internally, necessarily within a single transaction with the update) that the current values are as expected, and if not abort with an error like "the ideal is not in the expected phase".

**Comment by Mikolaj** [comment]
LGTM, though I don't understand some of the details.

File: src/Frontend/Page/Overview.hs (hunk 1 above)
Comment (fisx) [inline src/Frontend/Page/Overview.hs]: > Perhaps Position? There is already ` data IdeaLocation`. Also, we don't want to mix it up with Space.

you're right, and both `location` and `position` are still very general and thus not very meaningful.  `WhatList`?

i'm really fond of the `WhoWhen` type, but i'm not sure if anybody can share this enthusiasm.  :)

dada!

File: src/Persistent/Pure.hs (hunk 2 above)
Comment (fisx) [inline src/Persistent/Pure.hs]: On Tue, Apr 12, 2016 at 06:02:27PM -0700, Mikolaj Konarski wrote:
> Date: Tue, 12 Apr 2016 18:02:27 -0700
> From: Mikolaj Konarski <notifications@github.com>
> To: liqd/aula <aula@noreply.github.com>
> Cc: fisx <mf@zerobuzz.net>
> Subject: Re: [liqd/aula] Misc fixes. (#358)
>
> > @@ -245,6 +246,9 @@ runPersistExcept :: PersistExcept -> ServantErr
> >  runPersistExcept (PersistError500 msg)            = err500 { errBody = cs msg }
> >  runPersistExcept (PersistError404 msg)            = err404 { errBody = cs msg }
> >  runPersistExcept (PersistErrorNotImplemented msg) = err500 { errBody = cs msg }
> > +runPersistExcept (UserLoginInUse li) =
> > +    err500 { errBody = "user login in use: " <> cs (show li) }
> > +    -- FIXME: what's a good status code for 'login in use'?
>
> 200? USers are know to reload pages and go back, etc. Otherwise, perhaps 403:  https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.4

we should make sure that page reload on forms does not happen, and i think we do (see #83).  so 403 is probably the right one.  will fix, thanks!

File: static/js/custom.js [hunk 4]
```diff
@@ -88,7 +87,10 @@ function addClass(el, cl) {
...
 
-function incrCommentVote(e) {
-    var node = e.parentNode.parentNode.childNodes[0];
-    node.data = Number(node.data) + 1;
+function handleLikeOrVote(e) {
+    // FIXME: this is a race condition: if we wait for 0 ms, the page
+    // will usually be reloaded before the POST request updating the
+    // score can be processed.
+    // FIXME: we are required to avoid reload!
+    setTimeout(function() { document.location.reload(true); }, 50);
```
Comment (np) [inline static/js/custom.js]: @fisx why did you removed the previous version?

Comment (fisx) [inline static/js/custom.js]: > @fisx why did you removed the previous version?

sorry (:

my solution wasn't as successful as i hoped, but it has the nice effect on chrome that it looks almost like it works, even when clicking erratically on both many times.  your solution was not supporting the case where the user changed their mind.

anyway we'll need something more sophisticated.  probably involving ajax.
